### PR TITLE
DOC: special: complete the docstring for `fresnel`

### PR DESCRIPTION
**Pull request by person142**
#### Reference issue

None

#### What does this implement/fix?

The docstring for `special.fresnel` is completed.

#### Additional information

None.

### CODE REVIEW

File: scipy/special/add_newdocs.py [hunk 1]
```diff
@@ -3073,25 +3073,64 @@ def add_newdoc(name, doc):
...
+    References
+    ----------
+    .. [dlmf] NIST Digital Library of Mathematical Functions
+              https://dlmf.nist.gov/7.2#iii
+
+    Examples
+    --------
+    >>> import scipy.special as sc
+
+    As x goes to infinity, S and C converge to 0.5.
```
Comment (tylerjereddy) [inline scipy/special/add_newdocs.py]: Is the `x` here the `z` in the integrals & parameters above? That seems to be the case from the formula you've linked to from DLMF. Maybe that's obvious enough, but it had me thinking for a bit.

**Review by tylerjereddy — COMMENTED** [review]
This is a huge improvement over the devdocs version of the same docstring & I checked for consistency with DLMF.

I added a minor comment about one thing I didn't quite get, but probably just because of lack of domain knowledge.

File: scipy/special/add_newdocs.py (hunk 1 above)
Comment (person142) [reply scipy/special/add_newdocs.py]: Oops, changed everything to use `z` and specify that the limit is taken along the real axis.

**Review by tylerjereddy — APPROVED** [review]

**Comment by tylerjereddy** [comment]
Thanks Josh

**Pull request merged**